x/btcstaking/keeper/power_dist_change.go

```diff
@@ -189,7 +205,13 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		if fpActiveBTCDels, ok := activeBTCDels[fpBTCPKHex]; ok {
 			// handle new BTC delegations for this finality provider
 			for _, d := range fpActiveBTCDels {
-				fp.AddBTCDel(d)
+				del := *d
+				if _, slashed := slashedBTCDels[d.MustGetStakingTxHash().String()]; slashed {
+					// If the new BTC delegation is slashed, set its voting power to 0
+					del.TotalSat = 0
+				}
+
+				fp.AddBTCDel(&del)
```

Comment:
Instead of zero-ising `TotalSat` and add this BTC delegation, how aboure we skip adding it if slashed? Basically a slashed BTC delegation shall never appear in the voting power cache